### PR TITLE
Create custom mode for error messages

### PR DESCRIPTION
**Pull request by rgrinberg**
Error messages that are too long, are displayed in their own buffer. We
make sure that this buffer has its own mode. This makes it easy to
customize it with hooks and keybindings.

This addressed my issue in #1294

### CODE REVIEW

File: flycheck.el [hunk 1]
```diff
@@ -4862,9 +4866,13 @@ In the latter case, show messages in the buffer denoted by
 variable `flycheck-error-message-buffer'."
   (when (and errors (flycheck-may-use-echo-area-p))
     (let ((messages (seq-map #'flycheck-error-format-message-and-id errors)))
-      (display-message-or-buffer (string-join messages "\n\n")
-                                 flycheck-error-message-buffer
-                                 'not-this-window))))
+      (let ((result (display-message-or-buffer (string-join messages "\n\n")
+                                               flycheck-error-message-buffer
+                                               'not-this-window)))
+        (if (window-live-p result)
```
Comment (cpitclaudel) [inline flycheck.el]: This could be a `when`

File: flycheck.el [hunk 2]
```diff
@@ -4862,9 +4866,13 @@ In the latter case, show messages in the buffer denoted by
...
+            (with-current-buffer (window-buffer result)
+              (when (not (eq major-mode 'flycheck-error-message-mode))
```
Comment (cpitclaudel) [inline flycheck.el]: This could be an `unless`, and I think the convention is to use `derived-mode-p`

File: flycheck.el [hunk 3]
```diff
@@ -4849,6 +4849,10 @@ The echo area may be used if the cursor is not in the echo area,
 and if the echo area is not occupied by minibuffer input."
   (not (or cursor-in-echo-area (active-minibuffer-window))))
 
+(define-derived-mode flycheck-error-message-mode fundamental-mode
+  "Flycheck error messages"
+  "Major mode for extended error messages")
```
Comment (cpitclaudel) [inline flycheck.el]: Missing `.` at the end of the docstring.
Also, should the mode derive from text-mode?

**Comment by rgrinberg** [comment]
> LGTM, with a few minor tweaks. Feel free to merge after that ^^

Thanks for the review. I addressed your comments, but I don't have merging rights. A maintainer will need to do that.

**Pull request merged**